rasa/nlu/tokenizers/tokenizer.py

```diff
@@ -0,0 +1,39 @@
+from typing import Text, List, Optional, Dict, Any
+
+from rasa.nlu.constants import (
+    MESSAGE_RESPONSE_ATTRIBUTE,
+    MESSAGE_TEXT_ATTRIBUTE,
+    CLS_TOKEN,
+)
+
+
+class Token(object):
+    def __init__(self, text, offset, data=None):
+        self.offset = offset
+        self.text = text
+        self.end = offset + len(text)
+        self.data = data if data else {}
+
+    def set(self, prop, info):
+        self.data[prop] = info
+
+    def get(self, prop, default=None):
+        return self.data.get(prop, default)
+
+
+class Tokenizer(object):
```

Comment:
is there a case when it is not a `Component`? Why don't we use chain inheritance?

Reply:
ah, ok, saw the other comment

Reply:
now it `Tokenizer` inherits from `Component`